enhancements/plain-k8s-bundles.md

```diff
@@ -0,0 +1,220 @@
+    - i.e. the GVK is neither provided or required
+  - **provided** if it belongs to the group version of an APIService in the bundle
+  - **required** if it DOES NOT belong to:
+    - any group version defined by an APIService or CRD in the bundle
+    - a `*.k8s.io` group
+- `nonResourceURLs` referenced in RBAC with a subject shared by the Deployment will be considered **native APIs**
+
+The dependence on explicit kinds in RBAC rules means that rules with wildcard types (i.e. `*`) are disallowed in `k8s+v1` bundles. This may seem like a burdensome restriction, but a nice corollary is that all `k8s+v1` bundles follow [the principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). The exception to the restriction is RBAC w/ subjects that do not match a ServiceAccount in the bundle's Deployments.
+
+#### Non-Inferrable
+
+The following are parts of a CSV that are NOT directly inferrable from plain kubernetes resources:
+
+- Name
+- Descriptors
+- Description
+- Display Name
+- Keywords
+- Labels
+- Links
+- Maintainers
+- Maturity
+- Provider
+- Selector
+- Icon
+- Runtime Metadata
+- MinKubeVersion
+- (Future) Optional APIs
+- Replaces
+- Version
+- InstallModes
+
+This non-inferrable data will be gathered through a mixture of resource annotations and an `olm.yaml` file in the `metadata` directory of the bundle.
+
+Ex. _olm.yaml_
+
+```yaml
+name: packageserver-v1.14.1
+maturity: alpha
+version: 0.14.1
+replaces: packageserver-v1.14.0
+displayName: Package Server
+description: Represents an Operator package that is available from a given CatalogSource which will resolve to a ClusterServiceVersion.
+minKubeVersion: 1.11.0
+keywords: ['packagemanifests', 'olm', 'packages']
+maintainers:
+- name: Red Hat
+email: openshift-operators@redhat.com
+provider:
+name: Red Hat
+links:
+- name: Package Server
+url: https://github.com/operator-framework/operator-lifecycle-manager/tree/master/pkg/package-server
+installModes:
+- type: OwnNamespace
+  supported: true
+- type: SingleNamespace
+  supported: true
+- type: MultiNamespace
+  supported: true
+- type: AllNamespaces
+  supported: true
+icon: my-icon-png-base64-encoded
+labels:
+  operated-by: packageserver
+```
+
+#### `opm` Support
+
+[`opm`](https://github.com/operator-framework/operator-registry#overview) must be updated to provide support for building `k8s+v1` bundles. This can be acheived with explicit flags or through CSV/olm.yaml file presence.
+
+### CSV Synthesis
+
+OLM will support `k8s+v1` bundles by synthesizing an CSV from its contents. Resources usually generated by a CSV (e.g. APIService and Service) will not be directly applied, but instead be projected onto a synthesized CSV.
+
+This intermediary step (creating a CSV internally to represent the kubernetes objects) is a delibrate decision in 4.5 to help focus and deliver the feature. This may be relaxed in future release versions.
```

Comment:
It might be nice to include a section here about validation. It seems like we could add to the static analysis that the verification tool is doing to see if you have provided enough "stuff" (deployments, etc.) to synthesize a CSV

To go along with that, have we defined what is required vs optional (both raw kube manifests and in the csv.yaml)?

Reply:
> It might be nice to include a section here about validation. It seems like we could add to the static analysis that the verification tool is doing to see if you have provided enough "stuff" (deployments, etc.) to synthesize a CSV

I didn't really think through how/if the static validation framework would be used for this. I'll see if we can add a validator for this.

> To go along with that, have we defined what is required vs optional (both raw kube manifests and in the csv.yaml)?

I think these resources are implied in the _inferrable_ and _non-inferrable_ sections above, but I'll add a more explicit list.

Reply:
sections added.